### PR TITLE
Fixed author/content extraction

### PR DESCRIPTION
**Pull request by abhishekvm**
----
http://redhat.sys-con.com/node/4068643
Before:
"author": "ManageEngine IT Matters"
"author_description": "By ManageEngine IT Matters"

Now:
"author": "Elizabeth White"
"author_description": "http://elizabethwhite.sys-con.com/",

----
http://www.sltrib.com/home/5252884-155/vivint-smart-home-links-with-best
- Fixed content extraction - Missing few paragraphs
- Added rules in BEST_ELEMENT_PER_DOMAIN and setToRemove

----
http://therivardreport.com/san-antonio-space-scientists-prepare-for-jupiter-contact-1/
- Fixed content extraction - Removed excessive contents
- Added rules in NODES_TO_REMOVE_PER_DOMAIN

### CODE REVIEW

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java [hunk 1]
```diff
@@ -862,6 +882,26 @@ protected String extractTopPrivateDomain(String url){
...
+     * www.airpr.com -> airpr
+     * airpr.com -> airpr
+     *
+     * @param domain {@link String}: Top Level domain name
+     * @return: Domain name without tld and `www.`
+     */
+    protected String extractDomainNameWithoutTld(String domain) {
+
+        if (domain != null) {
+            Matcher macher = DOMAIN_WITHOUT_TLD.matcher(domain);
```
Comment (skyshard) [inline src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: typo

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java [hunk 2]
```diff
@@ -274,9 +291,11 @@ public ArticleTextExtractor() {
...
+        setToRemove("feedback-prompt|story-footer|story-meta-footer|related-combined-coverage|visuallyhidden|ad_topjobs|slideshow-overlay__data|next-post-thumbnails|video-desc|related-links|^widget popular$|^widget marketplace$|^widget ad panel$|slideshowOverlay|^share-twitter$|^share-facebook$|^share-google-plus-1$|^inline-list tags$|^tag_title$|article_meta comments|^related-news$|^recomended$|^news_preview$|related--galleries|image-copyright--copyright|^credits$|^photocredit$|^morefromcategory$|^pag-photo-credit$|gallery-viewport-credit|^image-credit$|story-secondary$|carousel-body|slider_container|widget_stories|post-thumbs|^custom-share-links|socialTools|trendingStories|^metaArticleData$|jcarousel-container|module-video-slider|jcarousel-skin-tango|^most-read-content$|^commentBox$|^faqModal$|^widget-area|login-panel|^copyright$|relatedSidebar|shareFooterCntr|most-read-container|email-signup|outbrain|^wnStoryBodyGraphic|articleadditionalcontent|most-popular|shatner-box|story-supplement|global-magazine-recent|nocontent");
     }
 
+    private static final Pattern WAYFAIR_DATE_PATTERN = Pattern.compile("\"display_date\":\"([^\"]+)\"");
```
Comment (skyshard) [inline src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: Make this more extensible across different sites

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java [hunk 3]
```diff
@@ -261,6 +277,7 @@ public boolean hasHTMLTags(String text){
     private static final int MAX_LOG_LENGTH = 200;
     private static final int MIN_WEIGHT_TO_SHOW_IN_LOG = 10;
 
+    private static final Pattern DOMAIN_WITHOUT_TLD = Pattern.compile("(www\\.)?([^\\.]+).*");
```
Comment (skyshard) [inline src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java:282]: Does this work on domains with multiple subdomains

**Review by skyshard — CHANGES_REQUESTED** [review]

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java (hunk 3 above)
Comment (abhishekvm) [reply src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java:282]: In fact no. But the regex used in extractDomainNameWithoutTld accepts the privateTopLevelDomain which doesn't have subdomains in it.

Comment (abhishekvm) [reply src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java:282]: Adding necessary comment in method documentation. Dropping this comment

**Comment by abhishekvm** [comment]
@skyshard Please recheck the suggested changes.

**Pull request merged**